system_tests/seq_filter_test.go

```diff
@@ -0,0 +1,112 @@
+package arbtest
+
+import (
+	"context"
+	"errors"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/arbitrum_types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/params"
+
+	"github.com/offchainlabs/nitro/arbos"
+	"github.com/offchainlabs/nitro/arbos/arbosState"
+	"github.com/offchainlabs/nitro/arbos/arbostypes"
+	"github.com/offchainlabs/nitro/arbos/l1pricing"
+	"github.com/offchainlabs/nitro/util/arbmath"
+)
+
+func TestSequencerTxFilter(t *testing.T) {
+	t.Parallel()
+
+	builder, header, txes, hooks, cleanup := setupSequencerFilterTest(t, false)
+	defer cleanup()
+
+	block, err := builder.L2.ExecNode.ExecEngine.SequenceTransactions(header, txes, hooks)
+	if block != nil {
+		t.Fatal("block shouldn't be generated when all txes have failed")
+	}
+	Require(t, err) // There shouldn't be any error in block generation
+	if len(hooks.TxErrors) != 2 {
+		t.Fatalf("expected 2 tx errors, found: %d", len(hooks.TxErrors))
+	}
+	for _, err := range hooks.TxErrors {
+		if err.Error() != state.ErrArbTxFilter.Error() {
+			t.Fatalf("expected ErrArbTxFilter, found: %s", err.Error())
+		}
+	}
+}
+
+func TestSequencerBlockFilter(t *testing.T) {
+	t.Parallel()
+
+	builder, header, txes, hooks, cleanup := setupSequencerFilterTest(t, true)
+	defer cleanup()
+
+	block, err := builder.L2.ExecNode.ExecEngine.SequenceTransactions(header, txes, hooks)
```

Comment:
I'd like to see the block failing with a filtered transaction and not failing if we're using the same hook but not trying a filtered transaction

Reply:
I've added the tests for these